modules/vmseries/variables.tf

```diff
@@ -17,14 +17,21 @@ variable "network_interfaces" {
   description = <<-EOF
   List of the network interface specifications.
   Available options:
-  - `subnetwork`             - (Required|string) Self-link of a subnetwork to create interface in.
-  - `private_ip_name`        - (Optional|string) Name for a private address to reserve.
-  - `private_ip`             - (Optional|string) Private address to reserve.
-  - `create_public_ip`       - (Optional|boolean) Whether to reserve public IP for the interface. Ignored if `public_ip` is provided. Defaults to 'false'.
-  - `public_ip_name`         - (Optional|string) Name for a public address to reserve.
-  - `public_ip`              - (Optional|string) Existing public IP to use.
-  - `public_ptr_domain_name` - (Optional|string) Existing public PTR name to use.
-  - `alias_ip_ranges`        - (Optional|list) List of objects that define additional IP ranges for an interface, as specified [here](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#ip_cidr_range)
+  - `subnetwork`                  - (Required|string) Self-link of a subnetwork to create interface in.
+  - `stack_type`                  - (Optional|string) IP stack to use: IPV4_ONLY (default) or IPV4_IPV6.
```

Comment:
I don't think we should do that, because:
- `create_..._ip` is not about if to use IP version or not. It is actually used to reserve an address object in GCP. If there is no `create_..._ip` ephemeral IP can still be provided to a VM
- we expose `stack_type` in other modules, making some abstraction in this module makes no sense to me 
- "explicit is better than implicit" Zen rule

Reply:
Sorry, had an edited comment that I forgot to save - I meant \*ipv6\* options in general (not only the create* ones).
One of the reasons why I've raised it is that it's not checked consistently - ie. it's being checked when creating `private_ipv6`, but not for `public_ipv6` - while it still should be set to `IPV4_IPV6`.
Even if left as is, I think it should be documented in some way in interface options descriptions.